coffeecard/CoffeeCard.Library/Services/v2/TokenService.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using CoffeeCard.Common.Errors;
+using CoffeeCard.Library.Persistence;
+using CoffeeCard.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace CoffeeCard.Library.Services.v2;
+
+public class TokenService : ITokenService
+{
+    private readonly CoffeeCardContext _context;
+    private readonly IHashService _hashService;
+
+    public TokenService(CoffeeCardContext context, IHashService hashService)
+    {
+        _context = context;
+        _hashService = hashService;
+    }
+
+    public async Task<string> GenerateMagicLink(User user)
```

Comment:
Naming issue, this does not generate a magic link, but a token

Reply:
Renamed to `GenerateMagicLinkToken` (token of type magic link). https://github.com/AnalogIO/analog-core/pull/289/commits/b051b7c608b57e03e8c8684392b2dbbdcda3a950